### PR TITLE
Custom Objection Support

### PR DESCRIPTION
**Pull request by cents02**
This will enable the server to support custom objections.

### CODE REVIEW

File: server/network/aoprotocol.py [hunk 1]
```diff
@@ -430,8 +430,10 @@ def net_cmd_ms(self, args):
             return
         if sfx_delay < 0:
             return
-        if button not in (0, 1, 2, 3, 4):
-            return
+        if '4' in button and "<and>" not in button:
```
Comment (oldmud0) [inline server/network/aoprotocol.py:433]: Could you explain this logic?

File: server/network/aoprotocol.py [hunk 2]
```diff
@@ -430,8 +430,10 @@ def net_cmd_ms(self, args):
...
+            if not button.isdigit():
+                print(button)
```
Comment (oldmud0) [inline server/network/aoprotocol.py:435]: Extraneous print statement.

File: server/network/aoprotocol.py [hunk 3]
```diff
@@ -430,8 +430,10 @@ def net_cmd_ms(self, args):
             return
         if sfx_delay < 0:
             return
-        if button not in (0, 1, 2, 3, 4):
-            return
+        if '4' in button and "<and>" not in button:
+            if not button.isdigit():
```
Comment (oldmud0) [inline server/network/aoprotocol.py:434]: Clause can be folded into above if-statement.

File: server/network/aoprotocol.py [hunk 4]
```diff
@@ -341,29 +342,28 @@ def net_cmd_ms(self, args):
...
-                self.ArgType.INT, self.ArgType.INT, self.ArgType.INT,
-                self.ArgType.INT, self.ArgType.STR_OR_EMPTY, self.ArgType.INT,
-                self.ArgType.INT, self.ArgType.INT):
+                args, self.ArgType.STR, self.ArgType.STR_OR_EMPTY,#msg_type, pre
+                self.ArgType.STR, self.ArgType.STR, self.ArgType.STR,#folder, anim, text
+                self.ArgType.STR, self.ArgType.STR, self.ArgType.INT,# pos, sfx, anim_type
+                self.ArgType.INT, self.ArgType.INT, self.ArgType.INT_OR_STR,#cid, sfx_delay, button
+                self.ArgType.INT, self.ArgType.INT, self.ArgType.INT,#evidence, flip, ding
+                self.ArgType.INT, self.ArgType.STR_OR_EMPTY, self.ArgType.INT, #color, showname, charid_pair
+                self.ArgType.INT, self.ArgType.INT): #offset_pair, nonint_pre
```
Comment (oldmud0) [inline server/network/aoprotocol.py:366]: Please leave a space before and after the `#`.

File: server/network/aoprotocol.py [hunk 5]
```diff
@@ -150,7 +151,7 @@ def validate_net_cmd(self, args, *types, needs_auth=True):
             return False
         if len(args) != len(types):
             return False
-        for i, arg in enumerate(args):
+        for i, arg in enumerate(args):
```
Comment (oldmud0) [inline server/network/aoprotocol.py:154]: Trailing whitespace.

File: server/network/aoprotocol.py (hunk 1 above)
Comment (cents02) [reply server/network/aoprotocol.py:433]: Yes, it checks whether the packet is valid. e.g:
"4<and>abc.gif" -> valid as both 4 and <and> exist
"4abcdef" -> invalid format as 4 exist but "<and>" doesn't
"4" -> valid as its a custom objection

File: server/network/aoprotocol.py (hunk 3 above)
Comment (cents02) [reply server/network/aoprotocol.py:434]: Fair enough

File: server/network/aoprotocol.py (hunk 2 above)
Comment (cents02) [reply server/network/aoprotocol.py:435]: Yeah fair enough

File: server/network/aoprotocol.py (hunk 4 above)
Comment (cents02) [reply server/network/aoprotocol.py:366]: Ok forgot to maintain the artistic standard.

**Pull request closed without merging**